WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderNavigationLogger.kt

```
@@ -0,0 +1,41 @@
+package com.woocommerce.android.ui.orders
+
+import android.annotation.SuppressLint
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavController
+import androidx.navigation.NavGraph
+import com.automattic.android.tracks.crashlogging.CrashLogging
+import kotlinx.coroutines.flow.StateFlow
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class OrderNavigationLogger @Inject constructor(private val crashLogging: CrashLogging) {
+
+    @SuppressLint("RestrictedApi")
+    fun logBackStack(navController: NavController, actionDescription: String) {
+        val backStackEntries: StateFlow<List<NavBackStackEntry>> = navController.currentBackStack
+        val backStackDescriptions = backStackEntries.value.joinToString(separator = ", ") { entry ->
+            val destination = entry.destination
+            val id = destination.id
+            val label = destination.label ?: "No label"
+            val className = destination.javaClass.simpleName
+            "ID=$id, Label=$label, Class=$className"
+        }
+
+        val rootGraph = navController.graph as? NavGraph
+        val startDestination = rootGraph?.findNode(rootGraph.startDestinationId)
+        val startDestinationDetails = startDestination?.let { destination ->
+            val id = destination.id
+            val label = destination.label ?: "No label"
+            val className = destination.javaClass.simpleName
+            "ID=$id, Label=$label, Class=$className"
+        } ?: "No start destination"
+
+        val logMessage = "$actionDescription: NavGraph=${rootGraph?.displayName}, " +
+            "StartDestination={$startDestinationDetails}, " +
+            "BackStackEntries=[$backStackDescriptions]"
+
+        crashLogging.recordEvent(logMessage)
```

⚠️ I'm a bit worried this might consume our quota in Sentry - AFAIU this will send events from all devices, even those which do not crash. What if we logged it only into the app logs?

---

I understandable your point of view. I leveraged this approach because I was hoping to only send these events when the crash occurs. This is the description of the method. 

```
    /**
     * Records a breadcrumb during the app lifecycle but doesn't report an event. This basically
     * adds more context for the next reports created and sent by [sendReport] or an unhandled
     * exception report.
     *
     * @param[message] The message to attach to a breadcrumb
     * @param[category] An optional category
     */
    fun recordEvent(
        message: String,
        category: String? = null
    )
```

However, I am not against utilization of the app logs if that works better because I have never used this `recordEvent` before so I am thinking I would also need to throw an exception or something to simulate a report for verification purposes.

---

I used `WooLog` instead. Thanks for the tip.

---

> I understandable your point of view. I leveraged this approach because I was hoping to only send these events when the crash occurs. This is the description of the method.

Ohh, very good point. Sorry, I missed that. I thought the method with this behavior was called recordBreadcrumb, but it is called recordEvent. In that case, I believe using recordEvent would be fine. Having said that, especially considering we'll remove this code after we fix this bug => let's use what we have in the code right now, which is WooLog.